### ISSUE TITLE
hoa-light object share the speaker definition 

### ISSUE THREAD

**Issue by chikashimiyama** [pd-externals/hoalight#11]
Speaker definition object holds the information of speaker setup. This object is currently only used for vbap but this can be theoretically used also by the hoa-light object

**Comment by chikashimiyama** [comment]
> I imagine this may involve changes to the vbap code which allow for exporting and reading the speaker array. Maybe this needs to be done via some sort of getter/setter interface?
by @danomatika

**Comment by chikashimiyama** [comment]
I thought [define_loudspeakers] object behaves as a source of truth for all vbap objects in the patch so that I don't have to send "define_loudspeakers" messages to all vbap objects (and eventually hoalight objects) but the help file of define_loudspeakers just confuses me.

<img width="940" alt="Screenshot 2021-03-29 at 11 27 11" src="https://user-images.githubusercontent.com/3512188/112816485-c1b55700-9081-11eb-8d5e-f51cfb8192b8.png">

It looks like the purpose of [define_loudspeakers] object is to tweak the triangle selection. 

In other words, how can I associate a [vbap] object and [define_loudspeakers] object properly?
Is that global scope like singleton (then that's a big problem)

**Comment by chikashimiyama** [comment]
@danomatika 

I think now I understand how to use [define_loudspeakers] object.


<img width="438" alt="Screenshot 2021-03-29 at 11 43 06" src="https://user-images.githubusercontent.com/3512188/112818695-1659d180-9084-11eb-9964-da22e92075c4.png">

it outputs **loudspeaker-matrices**  when I send a bang to it and vbap object accepts these matrices. 
in this way, if I want to tweak the vbap algorithm using ls-triplets or ls-direction I just do it once, because loudspeaker-matrices also contain tweaking. 

so sharing the speaker position between vbap and hoalight should be done in another way and probably it doesn't worth the effort.

**Comment by danomatika** [comment]
Yeah, I was imagining that hoalight could use the loudspeaker-matrices output from [define_loudspeakers]. If it's too much work for now, then leave it as other things are more important.
